unattended_installer/install_functions/opendistro/common.sh

```diff
@@ -274,107 +274,86 @@ function restoreWazuhrepo() {
-        if [ "${sys_type}" == "yum" ]; then
-            eval "yum remove opendistroforelasticsearch -y ${debug}"
-            eval "yum remove elasticsearch* -y ${debug}"
-            eval "yum remove opendistro-* -y ${debug}"
-        elif [ "${sys_type}" == "zypper" ]; then
-            eval "zypper -n remove opendistroforelasticsearch elasticsearch* opendistro-* ${debug}"
-        elif [ "${sys_type}" == "apt-get" ]; then
-            eval "apt remove --purge ^elasticsearch* ^opendistro-* ^opendistroforelasticsearch -y ${debug}"
-        fi 
+    if [ -n "${AIO}" ] || [ ${uninstall_component_name} == "all" ] || [ ${uninstall_component_name} == "kibana" ] ; then
+        if [ -n "${kibanainstalled}" ] || [ -n "${kibana_remaining_files}" ]; then
+            uninstallkibana
+        else
+            logger "Kibana components were not found on the system so it was not uninstalled."
+        fi
     fi
 
-    if [[ ( -n "${elastic_remaining_files}" || -n "${elasticsearchinstalled}" ) && ( -n "${elasticsearch}" || -n "${AIO}" || -n "${uninstall}" ) ]]; then
-        eval "rm -rf /var/lib/elasticsearch/ ${debug}"
-        eval "rm -rf /usr/share/elasticsearch/ ${debug}"
-        eval "rm -rf /etc/elasticsearch/ ${debug}"
+    # Overwrite case
+    if [ -n "${overwrite}" ] && [ -n "${wazuh}" ]; then
+        uninstallmanager
+    fi
+    if [ -n "${overwrite}" ] && [ -n "${elasticsearch}" ]; then
+        uninstallelasticsearch
+    fi
+    if [ -n "${overwrite}" ] && [ -n "${kibana}" ]; then
+        uninstallkibana
     fi
 
-    if [[ -n "${filebeatinstalled}" && ( -n "${wazuh}" || -n "${AIO}" || -n "${uninstall}" ) ]]; then
-        logger -w "Removing Filebeat."
-        if [ "${sys_type}" == "yum" ]; then
-            eval "yum remove filebeat -y ${debug}"
-        elif [ "${sys_type}" == "zypper" ]; then
-            eval "zypper -n remove filebeat ${debug}"
-        elif [ "${sys_type}" == "apt-get" ]; then
-            eval "apt remove --purge filebeat -y ${debug}"
+    # rollBack case!
+    for component in "${componentList[@]}"; do
+        if [ "${component}" == "manager" ] || [ "${component}" == "elastichsearch" ] || [ "${component}" == "kibana" ] ; then
+            eval "uninstall$component"
         fi
-    fi
+    done
+
+    checkIfInstalled
 
-    if [[ ( -n "${filebeat_remaining_files}" || -n "${filebeatinstalled}" ) && ( -n "${wazuh}" || -n "${AIO}" || -n "${uninstall}" ) ]]; then
-        eval "rm -rf /var/lib/filebeat/ ${debug}"
-        eval "rm -rf /usr/share/filebeat/ ${debug}"
-        eval "rm -rf /etc/filebeat/ ${debug}"
+    if [ -n "${wazuhinstalled}" ] || [ -n "${wazuh_remaining_files}" ] || [ -n "${filebeatinstalled}" ] || [ -n "${filebeat_remaining_files}" ] || [ -n "${elasticsearchinstalled}" ] || [ -n "${elastic_remaining_files}" ] || [ -n "${kibanainstalled}" ] || [ -n "${kibana_remaining_files}" ]; then
+        logger -w "Some Wazuh components are still installed on this host."
```

Comment:
When could this casuistic happen?

Reply:
When the process of uninstalling a component is done. And there are others installed.

Consider if when uninstalling a component. And there is still another one installed that should:
  - be notified, hence the -w
  - And additionally, the repository configuration will not be deleted. This should only be installed whenever it is verified that no component is left installed.